src/part_file.cpp

```diff
@@ -118,8 +118,9 @@ namespace libtorrent
 
 			for (int i = 0; i < num_pieces; ++i)
 			{
-				int slot = read_uint32(ptr);
-				if (slot == 0xffffffff) continue;
+				std::uint32_t const uslot = read_uint32(ptr);
+				if (uslot == 0xffffffff) continue;
+				int const slot = int(uslot);
```

Comment:
I would imagine there's very little, if any at all, arithmetic on ``slot``. perhaps it could be an unsigned type (perhaps similar to ``allocation_slot`` in stack_allocator). making it a special type should probably be a separate patch though